### PR TITLE
ServiceNow input improve

### PR DESCRIPTION
**Pull request by r7-mgorny**
## Proposed Changes
ServiceNow input improve

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/ServiceNow:4.2.0. Step name: update_incident\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/servicenow:4.2.0 --debug run < tests/update_incident.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/ServiceNow:4.2.0. Step name: create_incident\n",
    "meta": {},
    "output": {
      "number": "INC0010564",
      "system_id": "ee7e6b24dbf4e450e9faa5730596192b"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/servicenow:4.2.0 --debug run < tests/create_incident.json
</summary>
</details>

### Validate

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 1772.8729999999998ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] For Markdown linting, install mdl: gem install mdl

[*] Validating python files for style...
./icon_servicenow/actions/create_ci/action.py:32:41: F541 f-string is missing placeholders
./icon_servicenow/triggers/incident_changed/trigger.py:6:1: F811 redefinition of unused 'time' from line 2
./icon_servicenow/triggers/incident_created/trigger.py:60:56: F541 f-string is missing placeholders
./icon_servicenow/triggers/incident_created/trigger.py:63:56: F541 f-string is missing placeholders
./icon_servicenow/triggers/incident_created/trigger.py:67:48: F541 f-string is missing placeholders
./icon_servicenow/util/request_helper.py:4:1: F401 'insightconnect_plugin_runtime.exceptions.ConnectionTestException' imported but unused
./icon_servicenow/util/request_helper.py:52:41: F541 f-string is missing placeholders
[FAIL] Fails flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
57 [0.. 50.. ]
Run started:2020-12-05 14:20:29.630463

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 1832
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

[*] Checking for typos with Misspell...
[SUCCESS] Passes Misspell check

```

<summary>
make validate
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: plugin.spec.yaml[67]: https://instance.servicenow.com
violation: plugin.spec.yaml[70]: https://instance.servicenow.com
violation: help.md[258]: https://example.service-now.com/api/now/table/cmn_location/US-East
violation: help.md[433]: https://example.service-now.com/api/now/table/cmn_location/US-East
violation: help.md[236]: https://example.service-now.com/api/now/table/sys_user_group/sysdomain
violation: help.md[410]: https://example.service-now.com/api/now/table/sys_user_group/sysdomain
violation: help.md[253]: https://example.service-now.com/api/now/table/cmdb_model/59d4c676db0fc700553363835b961949
violation: help.md[428]: https://example.service-now.com/api/now/table/cmdb_model/59d4c676db0fc700553363835b961949
violation: help.md[230]: https://example.service-now.com/api/now/table/alm_asset/ff5a6a55dbdef7002e12ff00ba9619d6
violation: help.md[403]: https://example.service-now.com/api/now/table/alm_asset/ff5a6a55dbdef7002e12ff00ba9619d6
[*] Plugin successfully validated!

----
[*] Total time elapsed: 20027.212ms

```

<summary>
icon-validate --all .
</summary>
</details>

### CODE REVIEW

**Review by ncramer-r7 — APPROVED** [review]

File: servicenow/icon_servicenow/actions/create_incident/action.py [hunk 1]
```diff
@@ -28,9 +46,9 @@ def run(self, params={}):
 
         sys_id = result.get("sys_id", "")
         number = result.get("number", "")
-        
+
         if sys_id is None:
-            raise PluginException(cause=f'Error: create_incident failed - no system_id returned.',
+            raise PluginException(cause='Error: create_incident failed - no system_id returned.',
```
Comment (mrinehart-r7) [inline servicenow/icon_servicenow/actions/create_incident/action.py:51]:
```suggestion
            raise PluginException(cause='Error: Create Incident failed - no system_id returned.',
```

File: servicenow/plugin.spec.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ products: ["insightconnect"]
 name: servicenow
 title: ServiceNow
 description: ServiceNow is a tool for managing incidents and configuration management. Using the ServiceNow plugin for Rapid7 InsightConnect, users can manage all aspects of incidents including creation, search, updates, as well as monitor them for changes
-version: 4.1.0
+version: 4.2.0
```
Comment (mrinehart-r7) [inline servicenow/plugin.spec.yaml:7]:
```suggestion
version: 5.0.0
```

File: servicenow/plugin.spec.yaml [hunk 3]
```diff
@@ -86,11 +86,90 @@ actions:
     title: Create Incident
     description: Create a new ServiceNow Incident record
     input:
-      create_data:
-        title: Create Data
-        description: JSON object containing the fields and values to create a new incident
-        type: object
+      caller:
+        title: Caller / User ID
```
Comment (mrinehart-r7) [inline servicenow/plugin.spec.yaml:90]:
```suggestion
        title: Caller/User ID
```

File: servicenow/plugin.spec.yaml [hunk 4]
```diff
@@ -148,11 +227,91 @@ actions:
...
         type: string
         required: true
-      update_data:
-        title: Update Data
-        description: JSON object containing the fields and values to update
-        type: object
-        required: true
+        example: ee7e6b24dbf4e450e9faa5730596192b
+      caller:
+        title: Caller / User ID
```
Comment (mrinehart-r7) [inline servicenow/plugin.spec.yaml:232]:
```suggestion
        title: Caller/User ID
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]
Version should be bumped to 5.0.0 and plugin regenerated due to input changes

**Review by mrinehart-r7 — APPROVED** [review]

**Review by ncramer-r7 — APPROVED** [review]

**Review by jschipp-r7 — APPROVED** [review]

**Pull request merged**